src/player_avatar.c

```diff
@@ -884,22 +884,26 @@ int PlayerAvatar_DistortionWorldState (PlayerAvatar * const playerAvatar)
     return state;
 }
 
-BOOL sub_0205F158 (PlayerAvatar * const playerAvatar)
+// these functions can be simplified, but they'll no longer match
+
+BOOL PlayerAvatar_DistortionGravityChanged (PlayerAvatar * const playerAvatar)
 {
-    int v0 = PlayerAvatar_DistortionWorldState(playerAvatar);
+    int state = PlayerAvatar_DistortionState(playerAvatar);
 
-    if ((v0 == 0) || (v0 == 1)) {
+    if (state == AVATAR_DISTORTION_STATE_NONE || state == AVATAR_DISTORTION_STATE_ACTIVE) {
         return FALSE;
     }
 
     return TRUE;
```

Comment:
Does this match?

```c
return (state == AVATAR_DISTORTION_STATE_NONE || state == AVATAR_DISTORTION_STATE_ACTIVE) == FALSE;
```

Or, if possible:

```c
return state != AVATAR_DISTORTION_STATE_NONE && state != AVATAR_DISTORTION_STATE_ACTIVE;
```

I'd recommend looking at the ASM here and fiddling with the branch-return sequence to see if you can find something that matches more cleanly.

Reply:
Did some fiddling and we determined that the feasible match here is:

```c
return PlayerAvatar_DistortionState(playerAvatar) > (u32)AVATAR_DISTORTION_STATE_FLOOR;
```

That's pretty smelly, so we'll keep this sentinel structure instead.